packages/web-features-figuring/src/baseline/index.ts

```diff
@@ -0,0 +1,228 @@
+import assert from "node:assert";
+
+import { Temporal } from "@js-temporal/polyfill";
+
+import { feature } from "../browser-compat-data/feature";
+import { Release } from "../browser-compat-data/release";
+import { browsers, highReleases, lowReleases } from "./core-browser-set";
+import { support } from "./support";
+import { Browser } from "../browser-compat-data/browser";
+import { query } from "../browser-compat-data";
+
+interface FeatureSelector {
+  compatKeys: [string, ...string[]];
+  checkAncestors: boolean;
+}
+
+interface SupportStatus {
+  compatKey?: string;
+  baseline: "low" | "high" | false;
+  baseline_low_date: string | null;
+  baseline_high_date: string | null;
+  support: Map<Browser, Release | undefined>;
+  toJSON: () => string;
+}
+
+export function computeBaseline(
+  featureSelector: FeatureSelector,
+): SupportStatus {
+  const { compatKeys } = featureSelector;
+  const keys = featureSelector.checkAncestors
+    ? compatKeys.flatMap(withAncestors)
+    : compatKeys;
+
+  const statuses = keys.map(calculate);
+
+  const baseline = minStatus(statuses);
+
+  let baseline_low_date = null;
+  if (baseline !== false) {
+    baseline_low_date = latestDate(
+      statuses.map((status) => status.baseline_low_date),
+    );
+  }
+
+  let baseline_high_date = null;
+  if (baseline === "high") {
+    baseline_high_date = latestDate(
+      statuses.map((status) => status.baseline_high_date),
+    );
+  }
+
+  return {
+    baseline,
+    baseline_low_date,
+    baseline_high_date,
+    support: minSupport(statuses.map((status) => status.support)),
+    toJSON: function () {
+      return jsonify(this);
+    },
+  };
+}
+
+function calculate(compatKey: string): SupportStatus {
+  const f = feature(compatKey);
+  const s = support(f, browsers);
+
+  const allReleases = f.supportedBy();
+  const isBaselineLow = lowReleases.every((r) => allReleases.includes(r));
+  const isBaselineHigh = highReleases.every((r) => allReleases.includes(r));
+
+  const baseline =
+    (isBaselineHigh ? "high" : false) || (isBaselineLow ? "low" : false);
+
+  let baseline_low_date = null;
+  let baseline_high_date = null;
+  if (isBaselineLow) {
+    const initialReleases = [...s.entries()].map(([, r]) => r);
+    const keystoneRelease = initialReleases
+      .sort((a, b) => {
+        assert(a !== undefined);
+        assert(b !== undefined);
+        return a.date().getTime() - b.date().getTime();
+      })
+      .pop();
+
+    assert(keystoneRelease instanceof Release);
+    baseline_low_date = keystoneRelease.date().toISOString().slice(0, 10);
+
+    if (isBaselineHigh) {
+      const date = keystoneRelease.date();
+      // TODO: Replace with Temporal
+      date.setMonth(date.getMonth() + 30);
+      baseline_high_date = date.toISOString().slice(0, 10);
```

Comment:
I guess we should only compute `baseline_high_date` in one place. But if we're going to compute the main Baseline status as well, that really sugests that this package is where most of the computation should be, and not in https://github.com/web-platform-dx/web-features/blob/main/index.ts. We also want to be able to review the resulting `baseline_high_date` which we currently can't.

I think the model is that web-features-figuring is really our build script, which produces some artifacts (logs) that we commit, and some (the complete resulting JSON) that we don't.